build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchTestBasePlugin.java

```diff
@@ -235,13 +246,28 @@ private void configureImmutableCollectionsPatch(Project project) {
             .create(configurationName, config -> config.setCanBeConsumed(false));
         var deps = project.getDependencies();
         deps.add(configurationName, deps.project(Map.of("path", patchProject, "configuration", "patch")));
+
+        // If ElasticsearchJavaBasePlugin has specified a dependency on the entitlement bridge jar,
+        // then it needs to be added to the --patch-module=java.base command line option.
+        ConfigurationContainer configurations = project.getConfigurations();
+
         project.getTasks().withType(Test.class).matching(task -> task.getName().equals("test")).configureEach(test -> {
             test.getInputs().files(patchedFileCollection);
             test.systemProperty("tests.hackImmutableCollections", "true");
+
+            Configuration bridgeJarConfig = configurations.findByName("entitlementBridgeJar");
```

Comment:
why is half of this living here and half of the stuff in ElasticsearchJavaBase plugin?

Reply:
I'm moving it all to ElasticsearchTestBasePlugin.